api/src/main/java/io/strimzi/api/kafka/model/authentication/KafkaClientCustomAuthentication.java

```diff
@@ -0,0 +1,54 @@
+package io.strimzi.api.kafka.model.authentication;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+public class KafkaClientCustomAuthentication extends KafkaClientAuthentication {
+
+    public static final String TYPE_CUSTOM = "custom";
+    private String saslMechanism;
+    private String securityProtocol;
+    private String saslJaasConfig;
+    private String saslLoginCallbackHandlerClass;
+
+    @Override
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getType() {
+        return TYPE_CUSTOM;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getSaslMechanism() {
+        return saslMechanism;
+    }
+
+    public void setSaslMechanism(String saslMechanism) {
+        this.saslMechanism = saslMechanism;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getSecurityProtocol() {
+        return securityProtocol;
+    }
+
+    public void setSecurityProtocol(String securityProtocol) {
+        this.securityProtocol = securityProtocol;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getSaslJaasConfig() {
+        return saslJaasConfig;
+    }
+
+    public void setSaslJaasConfig(String saslJaasConfig) {
+        this.saslJaasConfig = saslJaasConfig;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getSaslLoginCallbackHandlerClass() {
```

Comment:
Out of curiosity ... what is the difference between `sasl.client.callback.handler.class` and `sasl.login.callback.handler.class` Do you need the other one to be configurable as well?

Reply:
not familiar with 'sasl.client.callback.handler.class', from kafka documentation this is what they say (which does not say much) 
    public static final String SASL_CLIENT_CALLBACK_HANDLER_CLASS = "sasl.client.callback.handler.class";
    public static final String SASL_CLIENT_CALLBACK_HANDLER_CLASS_DOC = "The fully qualified name of a SASL client callback handler class "
        + "that implements the AuthenticateCallbackHandler interface.";

    public static final String SASL_LOGIN_CALLBACK_HANDLER_CLASS = "sasl.login.callback.handler.class";
    public static final String SASL_LOGIN_CALLBACK_HANDLER_CLASS_DOC = "The fully qualified name of a SASL login callback handler class "
            + "that implements the AuthenticateCallbackHandler interface. For brokers, login callback handler config must be prefixed with "
            + "listener prefix and SASL mechanism name in lower-case. For example, "
            + "listener.name.sasl_ssl.scram-sha-256.sasl.login.callback.handler.class=com.example.CustomScramLoginCallbackHandler";